# $$Introduction$$

## Covid-19 22 January 2020 to 11 April 2020 data analysis

###  Data Information
- Number of rows 25353
- Number of columns 6
 - column_Name(Id)   Data_Type(int64) Records(25353)
 - column_Name(Province_State)   Data_Type(object) Records(10773 )
 - column_Name(Country_Region)   Data_Type(object) Records(25353)
 - column_Name(Date)   Data_Type(datetime64[ns]) Records(25353)
 - column_Name(ConfirmedCases)   Data_Type(float64) Records(25353)
 - column_Name(Fatalities)   Data_Type(float64) Records(25353)

## Necessery Liberary Import

In [1]:
import pandas as pd # Load data
import numpy as np # Scientific Computing
import seaborn as sns
import plotly
from plotly.offline import download_plotlyjs, init_notebook_mode,plot, iplot
init_notebook_mode(connected=True)
import plotly.graph_objs as go
import warnings # Ignore Warnings
warnings.filterwarnings("ignore")
sns.set() # Set Graphs Background

## Load Data

In [2]:
data = pd.read_csv('data/train (3).csv')
data.head()

Id Province_State Country_Region        Date  ConfirmedCases  Fatalities
0   1            NaN    Afghanistan  2020-01-22             0.0         0.0
1   2            NaN    Afghanistan  2020-01-23             0.0         0.0
2   3            NaN    Afghanistan  2020-01-24             0.0         0.0
3   4            NaN    Afghanistan  2020-01-25             0.0         0.0
4   5            NaN    Afghanistan  2020-01-26             0.0         0.0

## Information Of Data

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25353 entries, 0 to 25352
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              25353 non-null  int64  
 1   Province_State  10773 non-null  object 
 2   Country_Region  25353 non-null  object 
 3   Date            25353 non-null  object 
 4   ConfirmedCases  25353 non-null  float64
 5   Fatalities      25353 non-null  float64
dtypes: float64(2), int64(1), object(3)
memory usage: 1.2+ MB


## Unique Country

In [4]:
data['Country_Region'].unique()

array(['Afghanistan', 'Albania', 'Algeria', 'Andorra', 'Angola',
       'Antigua and Barbuda', 'Argentina', 'Armenia', 'Australia',
       'Austria', 'Azerbaijan', 'Bahamas', 'Bahrain', 'Bangladesh',
       'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin', 'Bhutan',
       'Bolivia', 'Bosnia and Herzegovina', 'Botswana', 'Brazil',
       'Brunei', 'Bulgaria', 'Burkina Faso', 'Burma', 'Burundi',
       'Cabo Verde', 'Cambodia', 'Cameroon', 'Canada',
       'Central African Republic', 'Chad', 'Chile', 'China', 'Colombia',
       'Congo (Brazzaville)', 'Congo (Kinshasa)', 'Costa Rica',
       "Cote d'Ivoire", 'Croatia', 'Cuba', 'Cyprus', 'Czechia', 'Denmark',
       'Diamond Princess', 'Djibouti', 'Dominica', 'Dominican Republic',
       'Ecuador', 'Egypt', 'El Salvador', 'Equatorial Guinea', 'Eritrea',
       'Estonia', 'Eswatini', 'Ethiopia', 'Fiji', 'Finland', 'France',
       'Gabon', 'Gambia', 'Georgia', 'Germany', 'Ghana', 'Greece',
       'Grenada', 'Guatemala', 'Guinea', 'Guine

## Unique country Count

In [5]:
data['Country_Region'].nunique()

184

- There have 184 Unique Country present

## Country Records Count

In [6]:
data['Country_Region'].value_counts()

US                4374
China             2673
Canada             972
France             891
United Kingdom     891
                  ... 
Gambia              81
Mexico              81
Djibouti            81
Czechia             81
Iran                81
Name: Country_Region, Length: 184, dtype: int64

## Maximum & Minimum Date 

In [7]:
print(data['Date'].min())
print(data['Date'].max())

2020-01-22
2020-04-11


- Start date 22 january 2020
- End date 11 april 2020

## Group Country & Select Largest 15 Country by ConfirmedCases

In [8]:
data_15 = data.groupby('Country_Region', as_index=False)['ConfirmedCases','Fatalities'].sum()
data_15 = data_15.nlargest(15,'ConfirmedCases')
data_15

Country_Region  ConfirmedCases  Fatalities
171              US       5135445.0    147545.0
36            China       5096274.0    182450.0
84            Italy       2661341.0    297444.0
156           Spain       2237252.0    200412.0
65          Germany       1728391.0     24491.0
61           France       1476739.0    122427.0
80             Iran       1184893.0     75219.0
175  United Kingdom        749434.0     72805.0
170          Turkey        408544.0      8310.0
161     Switzerland        393842.0     11029.0
90     Korea, South        369752.0      4758.0
16          Belgium        319318.0     23652.0
120     Netherlands        310043.0     27083.0
32           Canada        233554.0      4282.0
9           Austria        220500.0      3175.0

## Create Active/Recover Column

In [9]:
data_15['Active/Recover'] = data_15['ConfirmedCases'] - data_15['Fatalities']
data_15

Country_Region  ConfirmedCases  Fatalities  Active/Recover
171              US       5135445.0    147545.0       4987900.0
36            China       5096274.0    182450.0       4913824.0
84            Italy       2661341.0    297444.0       2363897.0
156           Spain       2237252.0    200412.0       2036840.0
65          Germany       1728391.0     24491.0       1703900.0
61           France       1476739.0    122427.0       1354312.0
80             Iran       1184893.0     75219.0       1109674.0
175  United Kingdom        749434.0     72805.0        676629.0
170          Turkey        408544.0      8310.0        400234.0
161     Switzerland        393842.0     11029.0        382813.0
90     Korea, South        369752.0      4758.0        364994.0
16          Belgium        319318.0     23652.0        295666.0
120     Netherlands        310043.0     27083.0        282960.0
32           Canada        233554.0      4282.0        229272.0
9           Austria        220500.0      3175.0        217325.0

## Bar Plot ConfirmedCases For New Data

In [10]:
p_15 = data_15['Country_Region']
q_15 = data_15['ConfirmedCases']

data1 = go.Bar(x=p_15,y=q_15,name='ConfirmedCases')

layout = go.Layout(title='Country_Region VS ConfirmedCases',xaxis_title='Country_Region',yaxis_title='ConfirmedCases')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- The Highest Number of ConfirmedCases in US
- The lowest Number of ConfirmedCases in Austria

## Bar Plot For Fatalities For New Data

In [11]:
p_15 = data_15['Country_Region']
q_15 = data_15['Fatalities']

data1 = go.Bar(x=p_15,y=q_15)

layout = go.Layout(title='Country_Region VS Deaths',xaxis_title='Country_Region',yaxis_title='Deaths')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- The Highest Number of Deaths in Italy
- The lowest Number of Deaths in Austria

## Bar Plot For Active/Recover For New Data

In [12]:
p_15 = data_15['Country_Region']
q_15 = data_15['Active/Recover']

data1 = go.Bar(x=p_15,y=q_15)

layout = go.Layout(title='Country_Region VS Active/Recover',
                   xaxis_title='Country_Region',yaxis_title='Active/Recover')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- The Highest Number of Active/Recover in US
- The lowest Number of Active/Recover in Austria

## Country_Region Vs ConfirmedCases & Deaths for 15 Country

In [13]:
p_15 = data_15['Country_Region']
q_15 = data_15['Active/Recover']
z_15 = data_15['Fatalities']
m_15 = data_15['ConfirmedCases']

data1 = go.Bar(x=p_15,y=m_15,name='ConfirmedCases')
data2 = go.Bar(x=p_15,y=q_15,name='Active/Recover')
data3 = go.Bar(x=p_15,y=z_15,name='Deaths')

layout = go.Layout(barmode='group',title='ConfirmedCases & Active/Recover & Deaths VS Country_Region',
                   xaxis_title='Country_Region',
                   yaxis_title='ConfirmedCases & Active/Recover & Deaths')
data=[data1,data2,data3]
fig = go.Figure(data,layout)

iplot(fig)

## Scatter Plot For Country VS ConfirmedCases For New Data

In [14]:
p_15 = data_15['Country_Region']
q_15 = data_15['ConfirmedCases']

data1 = go.Scatter(x=p_15,y=q_15,mode='markers')

layout = go.Layout(title='Country_Region VS ConfirmedCases',
                   xaxis_title='Country_Region',yaxis_title='ConfirmedCases')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- The Highest Number of ConfirmedCases in US.
- The lowest Number of ConfirmedCases in Austria.

## Scatter Plot For Country VS Active/Recover For New Data

In [15]:
p_15 = data_15['Country_Region']
q_15 = data_15['Active/Recover']

data1 = go.Scatter(x=p_15,y=q_15,mode='markers')

layout = go.Layout(title='Country_Region VS Active/Recover',
                   xaxis_title='Country_Region',yaxis_title='Active/Recover')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- The Highest Number of Active/Recover in US.
- The lowest Number of Active/Recover in Austria.

## Scatter Plot For Country VS Deaths For New Data

In [16]:
p_15 = data_15['Country_Region']
q_15 = data_15['Fatalities']

data1 = go.Scatter(x=p_15,y=q_15,mode='markers')

layout = go.Layout(title='Country_Region VS Deaths',
                   xaxis_title='Country_Region',yaxis_title='Deaths')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- The Highest Number of Deaths in Italy.
- The lowest Number of Deaths in Austria.

## Pie Chart For ConfirmedCases For New Data

In [17]:
pp = data_15['Country_Region']
qq = data_15['ConfirmedCases']

data = go.Pie(labels=pp,values=qq, 
              hoverinfo='label+percent', textinfo='value')

layout = go.Layout(title='Country_Region VS ConfirmedCases')
data =[data]
fig = go.Figure(data,layout)
iplot(fig)

- The Highest Number of ConfirmedCases 23% in US & China .
- The lowest Number of ConfirmedCases 1% in Belgium, Netherlands, Canada & Austria.

## Pie Chart For Deaths For New Data

In [18]:
pp = data_15['Country_Region']
qq = data_15['Fatalities']

data = go.Pie(labels=pp,values=qq, 
              hoverinfo='label+percent', textinfo='value')

layout = go.Layout(title='Country_Region VS Fatalities')
data =[data]
fig = go.Figure(data,layout)
iplot(fig)

- The Highest Number of Deaths 25% in Italy.
- The lowest Number of Deaths 0.2% in Austria.

## The Date Convert Into YYYY-MM-DD Format

In [19]:
data1 = pd.read_csv('data/train (3).csv')
data1['Date'] = pd.to_datetime(data1['Date'])

## Create New Dataset For Individual Date

In [20]:
data_81 = data1.groupby('Date', as_index=False)['ConfirmedCases','Fatalities'].sum()
data_81 = data_81.nlargest(81,'ConfirmedCases')
data_81

Date  ConfirmedCases  Fatalities
80 2020-04-11       1771337.0    108497.0
79 2020-04-10       1691542.0    102519.0
78 2020-04-09       1595174.0     95449.0
77 2020-04-08       1510928.0     88332.0
76 2020-04-07       1425923.0     81859.0
..        ...             ...         ...
4  2020-01-26          2113.0        56.0
3  2020-01-25          1432.0        42.0
2  2020-01-24           939.0        26.0
1  2020-01-23           653.0        18.0
0  2020-01-22           554.0        17.0

[81 rows x 3 columns]

## Create Active/Recover Column for Individual date

In [21]:
data_81['Active/Recover'] = data_81['ConfirmedCases'] - data_81['Fatalities']
data_81

Date  ConfirmedCases  Fatalities  Active/Recover
80 2020-04-11       1771337.0    108497.0       1662840.0
79 2020-04-10       1691542.0    102519.0       1589023.0
78 2020-04-09       1595174.0     95449.0       1499725.0
77 2020-04-08       1510928.0     88332.0       1422596.0
76 2020-04-07       1425923.0     81859.0       1344064.0
..        ...             ...         ...             ...
4  2020-01-26          2113.0        56.0          2057.0
3  2020-01-25          1432.0        42.0          1390.0
2  2020-01-24           939.0        26.0           913.0
1  2020-01-23           653.0        18.0           635.0
0  2020-01-22           554.0        17.0           537.0

[81 rows x 4 columns]

## Scatter Plot Date VS ConfirmedCases For New Dataset

In [22]:
p_81 = data_81['Date']
q_81 = data_81['ConfirmedCases']

data1 = go.Scatter(x=p_81,y=q_81)

layout = go.Layout(title='Date VS ConfirmedCases',
                   xaxis_title='Date',yaxis_title='ConfirmedCases')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- Day By Day Increase The ConfirmedCases

## Scatter Plot Date VS Active/Recover For New Dataset

In [23]:
p_81 = data_81['Date']
q_81 = data_81['Active/Recover']

data1 = go.Scatter(x=p_81,y=q_81)

layout = go.Layout(title='Date VS Active/Recover',
                   xaxis_title='Date',yaxis_title='Active/Recover')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- Day By Day Increase The Active/Recover

## Scatter Plot Date VS Deaths For New Dataset

In [24]:
p_81 = data_81['Date']
q_81 = data_81['Fatalities']

data1 = go.Scatter(x=p_81,y=q_81,mode='markers')

layout = go.Layout(title='Date VS Fatalities',
                   xaxis_title='Date',yaxis_title='Fatalities')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- Day By Day Increase The Deaths

## ConfirmedCases & Active/Recover & Deaths By Date

In [25]:
p_81 = data_81['Date']
q_81 = data_81['ConfirmedCases']
z_81 = data_81['Active/Recover']
m_81 = data_81['Fatalities']

trace0 = go.Scatter(x=p_81,y=q_81,mode='markers')
trace1 = go.Scatter(x=p_81,y=z_81,mode='markers')
trace2 = go.Scatter(x=p_81,y=m_81,mode='markers')

layout = go.Layout(title='Total ConfirmedCases & Active/Recover & Deaths By Date',
                   xaxis_title='Date',yaxis_title='ConfirmedCases Active/Recover & Deaths')
data=[trace0,trace1,trace2]
fig = go.Figure(data,layout)

iplot(fig)

- ConfirmedCases & Active/Recover highly increases after 18th March
- Deaths are increase after 29th March.

## Bar Plot Date VS ConfirmedCases For New Dataset

In [26]:
p_81 = data_81['Date']
q_81 = data_81['ConfirmedCases']

data1 = go.Bar(x=p_81,y=q_81)

layout = go.Layout(title='Date VS ConfirmedCases',
                   xaxis_title='Date',yaxis_title='ConfirmedCases')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- Day By Day Increase The ConfirmedCases.

## Bar Plot Date VS Deaths For World

In [27]:
p_81 = data_81['Date']
q_81 = data_81['Fatalities']

data1 = go.Bar(x=p_81,y=q_81)

layout = go.Layout(title='Date VS Deaths',
                   xaxis_title='Date',yaxis_title='Deaths')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- Day By Day Increase The Deaths.

## Line Plot Date VS ConfirmedCases For World

In [28]:
p_81 = data_81['Date']
q_81 = data_81['ConfirmedCases']

data1 = go.Scatter(x=p_81,y=q_81)

layout = go.Layout(title='Date VS ConfirmedCases',
                   xaxis_title='Date',yaxis_title='ConfirmedCases')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- Day By Day Increase The ConfirmedCases.

## Create New Dataset For Individual Date & Individual Country

In [29]:
data2 = pd.read_csv('data/train (3).csv')
data_all = data2.groupby(['Date','Country_Region'], as_index=False)['ConfirmedCases','Fatalities'].sum()
data_all

Date      Country_Region  ConfirmedCases  Fatalities
0      2020-01-22         Afghanistan             0.0         0.0
1      2020-01-22             Albania             0.0         0.0
2      2020-01-22             Algeria             0.0         0.0
3      2020-01-22             Andorra             0.0         0.0
4      2020-01-22              Angola             0.0         0.0
...           ...                 ...             ...         ...
14899  2020-04-11             Vietnam           258.0         0.0
14900  2020-04-11  West Bank and Gaza           268.0         2.0
14901  2020-04-11      Western Sahara             4.0         0.0
14902  2020-04-11              Zambia            40.0         2.0
14903  2020-04-11            Zimbabwe            14.0         3.0

[14904 rows x 4 columns]

## Create Active/Recover Column for Country &  date

In [30]:
data_all['Active/Recover'] = data_all['ConfirmedCases'] - data_all['Fatalities']
data_all

Date      Country_Region  ConfirmedCases  Fatalities  \
0      2020-01-22         Afghanistan             0.0         0.0   
1      2020-01-22             Albania             0.0         0.0   
2      2020-01-22             Algeria             0.0         0.0   
3      2020-01-22             Andorra             0.0         0.0   
4      2020-01-22              Angola             0.0         0.0   
...           ...                 ...             ...         ...   
14899  2020-04-11             Vietnam           258.0         0.0   
14900  2020-04-11  West Bank and Gaza           268.0         2.0   
14901  2020-04-11      Western Sahara             4.0         0.0   
14902  2020-04-11              Zambia            40.0         2.0   
14903  2020-04-11            Zimbabwe            14.0         3.0   

       Active/Recover  
0                 0.0  
1                 0.0  
2                 0.0  
3                 0.0  
4                 0.0  
...               ...  
14899           258.0  
14900           266.0  
14901             4.0  
14902            38.0  
14903            11.0  

[14904 rows x 5 columns]

## Select All United State Data From New Dataset

In [31]:
data_usa = data_all.query("Country_Region=='US'")
data_usa

Date Country_Region  ConfirmedCases  Fatalities  Active/Recover
171    2020-01-22             US             0.0         0.0             0.0
355    2020-01-23             US             0.0         0.0             0.0
539    2020-01-24             US             0.0         0.0             0.0
723    2020-01-25             US             0.0         0.0             0.0
907    2020-01-26             US             0.0         0.0             0.0
...           ...            ...             ...         ...             ...
14155  2020-04-07             US        396063.0     12717.0        383346.0
14339  2020-04-08             US        428889.0     14690.0        414199.0
14523  2020-04-09             US        461274.0     16473.0        444801.0
14707  2020-04-10             US        496372.0     18581.0        477791.0
14891  2020-04-11             US        526233.0     20458.0        505775.0

[81 rows x 5 columns]

## Scatter Plot Date VS ConfirmedCases For United State

In [32]:
p_usa = data_usa['Date']
q_usa = data_usa['ConfirmedCases']

data1 = go.Scatter(x=p_usa,y=q_usa,mode='markers')

layout = go.Layout(title='Total ConfirmedCases By Date For United State',
                   xaxis_title='Date',yaxis_title='ConfirmedCases')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- Day By Day Increase The Number of ConfirmedCases In US.

## Scatter Plot Date VS Deaths For United State

In [33]:
p_usa = data_usa['Date']
q_usa = data_usa['Fatalities']

data1 = go.Scatter(x=p_usa,y=q_usa,mode='markers')

layout = go.Layout(title='Total Fatalities By Date For United State',
                   xaxis_title='Date',yaxis_title='Fatalities')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- Day By Day Increase The Number of deaths In US.

## ConfirmedCases & Active/Recover & Deaths By Date For United State

In [34]:
p_usa = data_usa['Date']
q_usa = data_usa['ConfirmedCases']
z_usa = data_usa['Active/Recover']
m_usa = data_usa['Fatalities']

trace0 = go.Scatter(x=p_usa,y=q_usa,mode='markers')
trace1 = go.Scatter(x=p_usa,y=z_usa,mode='markers')
trace2 = go.Scatter(x=p_usa,y=m_usa,mode='markers')

layout = go.Layout(title='Total ConfirmedCases & Active/Recover & Deaths By Date',
                   xaxis_title='Date',yaxis_title='ConfirmedCases Active/Recover & Deaths')
data=[trace0,trace1,trace2]
fig = go.Figure(data,layout)

iplot(fig)

- ConfirmedCases & Active/Recover highly increases
- Deaths are increase after 9th April.

## Bar Plot Date VS ConfirmedCases For United State

In [35]:
p_usa = data_usa['Date']
q_usa = data_usa['ConfirmedCases']

data1 = go.Bar(x=p_usa,y=q_usa)

layout = go.Layout(title='Date VS ConfirmedCases US',
                   xaxis_title='Date',yaxis_title='ConfirmedCases')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- Day By Day Increase The Number of ConfirmedCases In US.

## Bar Plot Date VS Deaths For United State

In [36]:
p_usa = data_usa['Date']
q_usa = data_usa['Fatalities']

data1 = go.Bar(x=p_usa,y=q_usa)

layout = go.Layout(title='Date VS Deaths US',
                   xaxis_title='Date',yaxis_title='Deaths')
data=[data1]
fig = go.Figure(data,layout)

iplot(fig)

- Day By Day Increase The Number of Deaths In US.